src/clone-element.js

```diff
@@ -3,9 +3,9 @@ import { h } from './h';
 
 /**
  * Clones the given VNode, optionally adding attributes/props and replacing its children.
- * @param {VNode} vnode		The virtual DOM element to clone
- * @param {Object} props	Attributes/props to add when cloning
- * @param {VNode} rest		Any additional arguments will be used as replacement children.
+ * @param {import('./vnode').VNode} vnode The virtual DOM element to clone
```

Comment:
Oh true! I'll do that.

Reply:
FYI, Had to add VNode to the eslint no-unused-var exception

Reply:
I feel like we should be consistent with this where possible. The latest changes have a top level `import` here and all other places use `import` inside jsdoc.

Reply:
Although I think the top level imports are easier to understand for new contributors, I'm a bit worried about reducing usefulness of `no-unused-var-exception`. After thinking a bit more about it I think I'll actually prefer the inline imports. What are your thoughts about this? @developit @andrewiggins

Reply:
Yea, that's fair. Consistency would be good. I don't have a strong preference for either pattern but since Preact is not TypeScript, I like the clean separation of type-related "code" (i.e. JSDoc) and real code. Putting the unused import at the top of the file may seem like a bug to new contributors since no real code uses it.

Reply:
@marvinhagemeister I've gone ahead and reverted the change back to importing the type inside the JSDoc comment to be consistent with the rest of the changes in this PR. I suggest if we want to use ES6 imports purely for JSDoc comments, we can do that in another PR that applies it consistently across the codebase. Doing that we can evaluate what it does to the codebase and better consider the maintenance tradeoffs, etc.

Reply:
Agree 💯